src/lib.rs

```diff
@@ -391,24 +393,44 @@ fn bytes_range(offset: u64) -> String {
     format!("bytes={}-", offset)
 }
 
+#[derive(Debug)]
+pub enum SyncParam {
+    Local { path: String },
+    Bucket { bucket: String, path: String },
+}
+
+impl std::str::FromStr for SyncParam {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s3_format = Regex::new(r"^s3://+?(?P<bucket>[a-zA-Z0-9._-]+)/(?P<prefix>.*)$").unwrap();
```

Comment:
For our purposes I think we can match "anything that's not a `/`" for the bucket name.